src/global/install.rs

```diff
@@ -88,138 +61,117 @@ pub struct ScriptExecMapping {
+    eprintln!("run activation scripts are {:?}", activation_variables);
+
+    let activator = Activator::from_path(prefix.root(), ShellEnum::default(), Platform::current())
+        .into_diagnostic()
+        .unwrap();
+
+    eprintln!("Activation scripts are {:?}", activator.activation_scripts);
+
     for ScriptExecMapping {
         global_script_path,
         original_executable,
     } in mapped_executables
     {
-        let mut script = activation_script.clone();
-        shell
-            .run_command(
-                &mut script,
-                [
-                    format!(
-                        "\"{}\"",
-                        prefix.root().join(original_executable).to_string_lossy()
-                    )
-                    .as_str(),
-                    get_catch_all_arg(shell),
-                ],
-            )
-            .expect("should never fail");
-
-        if matches!(shell, ShellEnum::CmdExe(_)) {
-            // wrap the script contents in `@echo off` and `setlocal` to prevent echoing the
-            // script and to prevent leaking environment variables into the
-            // parent shell (e.g. PATH would grow longer and longer)
-            script = format!(
-                "@echo off\nsetlocal\n{}\nset exitcode=%ERRORLEVEL%\nendlocal\nexit %exitcode%",
-                script.trim()
-            );
-        }
-
-        let added_or_changed = if global_script_path.exists() {
-            match tokio_fs::read_to_string(global_script_path).await {
-                Ok(previous_script) if previous_script != script => AddedOrChanged::Changed,
-                Ok(_) => AddedOrChanged::Unchanged,
-                Err(_) => AddedOrChanged::Changed,
+        let exe = prefix.root().join(original_executable);
+        let path = prefix
+            .root()
+            .join(original_executable.parent().ok_or_else(|| {
+                miette::miette!(
+                    "Cannot find parent directory of '{}'",
+                    original_executable.display()
+                )
+            })?);
+        let metadata = ManifestMetadata::new(exe, path, Some(activation_variables.clone()));
+
+        let json_path = global_script_path.with_extension("json");
+
+        let global_script_path = {
+            #[cfg(windows)]
+            {
+                &global_script_path.with_extension("exe")
+            }
+            #[cfg(unix)]
+            {
+                global_script_path
             }
-        } else {
-            AddedOrChanged::Added
         };
 
-        if matches!(
-            added_or_changed,
-            AddedOrChanged::Changed | AddedOrChanged::Added
-        ) {
-            tokio_fs::write(&global_script_path, script)
+        // Check if an old bash script is present and remove it
+        let mut changed = if global_script_path.exists() && !is_binary(global_script_path)? {
+            tokio::fs::remove_file(global_script_path)
```

Comment:
```suggestion
            tokio_fs::remove_file(global_script_path)
```
We use the `fs_err` crate now.